test/javascripts/unit/lib/apply-reply-test.js

```diff
@@ -0,0 +1,117 @@
+import EmberObject from "@ember/object";
+import { discourseModule } from "discourse/tests/helpers/qunit-helpers";
+import { prepareReply } from "discourse/plugins/discourse-canned-replies/lib/apply-reply";
+import { test } from "qunit";
+
+discourseModule(
+  "Unit | Plugins | discourse-canned-replies | Lib | apply-reply",
+  function () {
+    test("prepareReply", function (assert) {
+      const expectedVariables = {
+        my_username: "heisenberg",
+        my_name: "Walter White",
+        original_poster_username: "mr_hide",
+        original_poster_name: "Dr. Henry Jekyll",
+        reply_to_username: "dracula",
+        reply_to_name: "Vlad",
+        last_poster_username: "frankenstein",
+        reply_to_or_last_poster_username: "dracula",
+      };
+
+      // there must be a more elegant way to do this
+      const fakeModel = EmberObject.extend({
+        user: EmberObject.extend({
+          username: expectedVariables.my_username,
+          name: expectedVariables.my_name,
+        }).create(),
+        topic: EmberObject.extend({
+          details: EmberObject.extend({
+            created_by: EmberObject.extend({
+              username: expectedVariables.original_poster_username,
+              name: expectedVariables.original_poster_name,
+            }).create(),
+          }).create(),
+          last_poster_username: expectedVariables.last_poster_username,
+        }).create(),
+        post: EmberObject.extend({
+          username: expectedVariables.reply_to_username,
+          name: expectedVariables.reply_to_name,
+        }).create(),
+      }).create();
```

Comment:
I think you can simplify it like this:

```suggestion
      const fakeModel = EmberObject.create({
        user: {
          username: expectedVariables.my_username,
          name: expectedVariables.my_name,
        },
        topic: {
          details: {
            created_by: {
              username: expectedVariables.original_poster_username,
              name: expectedVariables.original_poster_name,
            },
          },
          last_poster_username: expectedVariables.last_poster_username,
        },
        post: {
          username: expectedVariables.reply_to_username,
          name: expectedVariables.reply_to_name,
        },
      });
```